markymark/src/main/java/com/m2mobi/markymark/composable/TableBlock.kt

```diff
@@ -0,0 +1,350 @@
+    for (row in 2 until rowHeights.size) {
+        var y = outlineThickness + headerDividerThickness
+        for (i in 0 until row) {
+            if (i > 1) y += bodyDividerThickness
+            y += rowHeights.getValue(i)
+        }
+
+        drawRect(
+            color = style.bodyDividerStyle.color,
+            topLeft = Offset(x = 0F, y = y),
+            size = Size(width = size.width, height = bodyDividerThickness),
+        )
+    }
+}
+
+private fun DrawScope.drawHeaderDivider(style: TableDividerStyle, yOffset: Float) {
+    drawRect(
+        color = style.color,
+        topLeft = Offset(x = 0F, y = yOffset),
+        size = Size(
+            width = size.width,
+            height = style.thickness.toPx(),
+        )
+    )
+}
+
+private fun DrawScope.drawOutlines(style: TableDividerStyle) {
+    val thickness = style.thickness.toPx()
+
+    // left
+    drawRect(
+        color = style.color,
+        topLeft = Offset(
+            x = 0F,
+            y = size.height - thickness,
+        ),
+        size = Size(
+            width = size.width,
+            height = thickness,
+        ),
+    )
+
+    // top
+    drawRect(
+        color = style.color,
+        size = Size(
+            width = size.width,
+            height = thickness,
+        ),
+    )
+
+    // right
+    drawRect(
+        color = style.color,
+        size = Size(
+            width = thickness,
+            height = size.height,
+        ),
+    )
+
+    // bottom
+    drawRect(
+        color = style.color,
+        topLeft = Offset(
+            x = size.width - thickness,
+            y = 0F,
+        ),
+        size = Size(
+            width = thickness,
+            height = size.height,
+        ),
+    )
+}
+
+class TableBlockMeasurePolicy(
+    private val style: TableBlockStyle,
+) : MeasurePolicy {
+
+    var measuredColumnWidths by mutableStateOf(emptyMap<Int, Int>())
+    var measuredRowHeights by mutableStateOf(emptyMap<Int, Int>())
+
+    override fun MeasureScope.measure(measurables: List<Measurable>, constraints: Constraints): MeasureResult {
+        val columnWidths = mutableMapOf<Int, Int>()
+        val rowHeights = mutableMapOf<Int, Int>()
+        val placeableCells = mutableListOf<PlaceableTableCell>()
+
+        for (measurable in measurables) {
+            val spec = measurable.layoutId as TableCellSpec
+            columnWidths[spec.column] = maxOf(
+                columnWidths[spec.column] ?: style.cellMinWidth.roundToPx(),
+                measurable.maxIntrinsicWidth(constraints.maxHeight),
+            ).coerceAtMost(style.cellMaxWidth.roundToPx())
+            rowHeights[spec.row] = maxOf(
+                rowHeights[spec.row] ?: style.cellMinWidth.roundToPx(),
+                measurable.maxIntrinsicHeight(constraints.maxWidth),
+            ).coerceAtMost(style.cellMaxHeight.roundToPx())
+        }
+
+        for (measurable in measurables) {
+            val spec = measurable.layoutId as TableCellSpec
```

Comment:
noice

Reply:
Yeah its great that they allow for `Any` as the id. Let's you pull stunts like this which is super usefull to add extra info to composables for when you measure them.